cognito-post-authentication/src/test/java/no/unit/nva/cognito/PostAuthenticationHandlerTest.java

```diff
@@ -47,80 +48,125 @@
     @BeforeEach
     public void init() {
         customerApi = mock(CustomerApi.class);
-        userApi = mock(UserApi.class);
+        userApi = new UserApiMock();
         awsCognitoIdentityProvider = mock(AWSCognitoIdentityProvider.class);
         userService = new UserService(userApi, awsCognitoIdentityProvider);
         handler = new PostAuthenticationHandler(userService, customerApi);
     }
 
     @Test
-    public void handleRequestUpdatesUserPoolWithExistingUserWhenUserIsFound() {
-        UUID customerId = UUID.randomUUID();
-        prepareMocksWithExistingCustomer(customerId);
-        prepareMocksWithExistingUser(SAMPLE_FEIDE_ID);
+    public void handleRequestUsesExistingUserWhenUserIsFound() {
+        prepareMocksWithExistingCustomer();
+        prepareMocksWithExistingUser();
 
         Event requestEvent = createRequestEvent();
         final Event responseEvent = handler.handleRequest(requestEvent, mock(Context.class));
 
-        verify(userApi, times(0)).createUser(any());
-        verify(awsCognitoIdentityProvider).adminAddUserToGroup(any());
+        verify(awsCognitoIdentityProvider, times(2)).adminAddUserToGroup(any());
```

Comment:
maybe refactor these two checks in checkThatUserWasAddedtoCognitoGroup? it is hard to understand the intention behind the requirement that awsCognitoIdentityProvider has to be called twice. You could refactor that line only to 
a method called `checkCognitoWasCalledForReasonOneAndTwo` because when this test will fail the most probable thing is that we will just change the number.